code/modules/wiremod/components/math/trig.dm

```diff
@@ -0,0 +1,132 @@
+	. = ..()
+	if(.)
+		return
+
+	//The output is based on the override of do_calculation()
+	output.set_output(do_calculation(input.input_value))
+
+
+//Does your typical Sine, Cosine, and Tangent, as well as their inverses
+/obj/item/circuit_component/trig/trig
+	display_name = "Trigonometry"
+	display_desc = "A component capable of trigonometric functions."
+
+/obj/item/circuit_component/trig/trig/populate_options()
+	var/static/component_options = list(
+		COMP_TRIG_SINE,
+		COMP_TRIG_COSINE,
+		COMP_TRIG_TANGENT,
+		COMP_TRIG_ASINE,
+		COMP_TRIG_ACOSINE,
+		COMP_TRIG_ATANGENT
+	)
+	options = component_options
+
+/obj/item/circuit_component/trig/trig/do_calculation(value)
+
+	switch(current_option)
+		if(COMP_TRIG_ASINE)
+			return (value >= -1 && value <= 1) ? arcsin(value) * PI/180 : null //Converting to Radians
+		if(COMP_TRIG_ACOSINE)
+			return (value >= -1 && value <= 1) ? arccos(value) * PI/180 : null
+		if(COMP_TRIG_ATANGENT)
+			return arctan(value) * PI/180
+	value *= 180/PI //Convert to degrees, because apparently BYOND doesn't believe in the almighty radian
+	switch(current_option)
+		if(COMP_TRIG_COSINE)
+			return cos(value)
+		if(COMP_TRIG_SINE)
+			return sin(value)
+		if(COMP_TRIG_TANGENT)
+			return cos(value) == 0 ? null : tan(value)
+	return null //Safety check incase something goes wrong with the component options
+
+
+//Performs Secant, Cosecant, and Cotangent
+/obj/item/circuit_component/trig/adv_trig
+	display_name = "Advanced Trigonometry"
+	display_desc = "Following outstanding advancements in the field of Mathematics, NanoTrasen scientist have discovered how to take the reciprical of trignometric functions"
+
+
+/obj/item/circuit_component/trig/adv_trig/populate_options()
+	var/static/component_options = list(
+		COMP_TRIG_SECANT,
+		COMP_TRIG_COSECANT,
+		COMP_TRIG_COTANGENT
+	)
+	options = component_options
+
+/obj/item/circuit_component/trig/adv_trig/do_calculation(value)
+	value *= 180/PI //Switch to degrees again
+	switch(current_option)
+		if(COMP_TRIG_SECANT)
+			return cos(value) == 0 ? null : SEC(value)
+		if(COMP_TRIG_COSECANT)
+			return sin(value) == 0 ? null : CSC(value)
+		if(COMP_TRIG_COTANGENT)
+			return sin(value) == 0 ? null : (cos(value) * CSC(value)) //The define for COT uses 1/tan(x), which throws a divide by zero error when x = pi/2 + kpi where k is an integer
+	return null //Safety check incase something goes wrong with the component options
+
+
+//Hyperbolic Sine and Cosine
+/obj/item/circuit_component/trig/hyper_trig
+	display_name = "Hyperbolic Trigonometry"
+	display_desc = "This component makes all your trig calculations be based on hyperbolas and natural exponentials instead of circles"
+
+/obj/item/circuit_component/trig/hyper_trig/populate_options()
+	var/static/component_options = list(
+		COMP_TRIG_HYPERBOLIC_SINE,
+		COMP_TRIG_HYPERBOLIC_COSINE,
+		COMP_TRIG_AHYPERBOLIC_SINE,
+		COMP_TRIG_AHYPERBOLIC_COSINE,
+	)
+	options = component_options
```

Comment:
why not just assign all this bull to options instead of doing this pointless casting in a variable you won't use for anything

Reply:
The local component_options var was how every other component did populate options, so I figured that I would do the same for consistency, but I'll change it to just write to options directly to get rid of redundancy if you want.